### DISCUSSION TITLE
`vue-component-meta` queries

### DISCUSSION THREAD

**Discussion by jd-solanki** [vuejs/language-tools#1809] [Q&A] [answered]
Hi 👋🏻
I am building [Anu](https://anu-vue.netlify.app/). I want to render the API table at the end of component demos.

I used vue-component-meta but I have few queries with it:
1. Is it still WIP?
2. Does it work with `.tsx` components?
3. It includes `ref_for` & `ref_key` as props, can we exclude them & how?
4. I always get slots as empty: `slots: [],` (This is same for events)
5. I see there's description property on listed props, so how can I add description in my code so that I can get it via `vue-component-meta`

Regards.

**Comment by johnsoncodehk** [comment, marked as answer]
1. It's launch.
2. Yes.
3. You can filter it by check `global` property: https://github.com/johnsoncodehk/volar/blob/2758510a8e519563f16ce023c700ab98b8b65ad8/packages/vue-component-meta/tests/index.spec.ts#L20
4. The component should have `$slots`, `$emit` type. (https://github.com/johnsoncodehk/volar/blob/master/packages/vue-test-workspace/vue-component-meta/class-slots/component.vue) If component write by script setup, volar code gen will auto expose slots type from template.
5. You can wirte JSDoc to props: https://github.com/johnsoncodehk/volar/blob/2758510a8e519563f16ce023c700ab98b8b65ad8/packages/vue-test-workspace/vue-component-meta/reference-type-props/my-props.ts#L2-L4

**Reply by jd-solanki** [reply]
I will give it a try and will let you know if something goes wrong.

Thanks for this useful package ❤️

**Reply by jd-solanki** [reply]
Just a quick question, I have `defineCompoent` with `.tsx` extension and using setup function and there's no `slots` or `$slots` property 🤔 
![Screenshot 2022-09-04 at 8 53 11 PM](https://user-images.githubusercontent.com/47495003/188321061-44a82b84-ea2e-48f3-a4e2-34c51d23134e.png)

**Reply by johnsoncodehk** [reply]
@jd-solanki Yes `defineComponent` dons't have it, it's still in vue RFC(https://github.com/vuejs/rfcs/pull/192). You need to shims the expose type you self.

```ts
const Component = defineComponent({ ... });
export default Component as typeof Component & new () {
	$slots: { ... }
}
```

**Reply by jd-solanki** [reply]
Hi @johnsoncodehk 

It is really awesome ❤️ 

Here's what I have achieved so far:
![Screenshot 2022-09-08 at 12 01 01 PM](https://user-images.githubusercontent.com/47495003/189050779-5bb1b1a8-dd05-4c47-a0ed-0228bb2f7af1.png)


However, Does it parse the code statically? Because it can't generate accurate data for `variant`, `states` & `color` prop:
- [useProps](https://github.com/jd-solanki/anu/blob/main/packages/anu-vue/src/composables/useLayer.ts#L4) composable
- [useProps usage](https://github.com/jd-solanki/anu/blob/main/packages/anu-vue/src/components/alert/AAlert.tsx#L7-L14)
